tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/Scala3SyntaxSuite.scala

```diff
@@ -0,0 +1,30 @@
+package scala.meta.tests.parsers.dotty
+
+import scala.meta._
+import scala.meta.dialects.Scala3
+
+class Scala3SyntaxSuite extends BaseDottySuite {
+
+  test("given intOrd: Ord[Int] with \n{ def f(): Int = 1 }") {
+    assertEquals(
+      templStat(
+        "given intOrd: Ord[Int] with \n{ def f(): Int = 1 }"
+      ).syntax,
+      "given intOrd: Ord[Int] with \n{ def f(): Int = 1 }"
+    )
+    val Defn.Given(_, _, _, _, template) =
+      templStat(
+        "given intOrd: Ord[Int] with \n{ def f(): Int = 1 }"
+      )
+    assertEquals(template.syntax, "Ord[Int] with \n{ def f(): Int = 1 }")
+  }
+
+  test("enum C extends A with B { case D }") {
+    assertEquals(
+      templStat("enum C extends A with B { case D }").syntax,
+      "enum C extends A with B { case D }"
+    )
+    val q"enum $name $template" = q"enum C extends A with B { case D }"
```

Comment:
@bjaglin quasiquotes with enums seems to work. But I had some issues with givens, however those might caused by the fact that givens can take a lot of forms. We might need to work on that depending on how much quasiquotes are sued here.